### PR TITLE
Add capabilities API and provider awareness, fix client initialization

### PR DESCRIPTION
**Pull request by marians**
This is supposed to do what https://github.com/giantswarm/gsctl/pull/356 already did, but based on a refactored gsctl code base.

Also fixes client initialization for the `refactor-command-tree` branch.

### CODE REVIEW

**Comment by marians** [comment]
This is based on the refactoring PR https://github.com/giantswarm/gsctl/pull/360 and will also be merged into that one, as it prvides a necessary fix (removing the mandadory client initialization in `root.go`.)

File: capabilities/capabilities.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+// ReleaseProviderPair is a combination of a providr ('aws', 'azure', 'kvm) and
+// a release version number.
+type ReleaseProviderPair struct {
+	Provider       string
+	ReleaseVersion *semver.Version
+}
+
+// GetCapabilities returns the capabilities available in the current context
+func GetCapabilities(provider, releaseVersion string) ([]CapabilityDefinition, error) {
+	cap := []CapabilityDefinition{}
```
Comment (tobbbles) [inline capabilities/capabilities.go]:
```suggestion
	capability := []CapabilityDefinition{}
```

`cap` is a reserved word for checking the capacity of a slice

Comment (marians) [reply capabilities/capabilities.go]: Changed

**Comment by marians** [comment]
@tobbbles Fine now?

**Pull request merged**